src/App.tsx

```diff
@@ -1,39 +1,61 @@
-import React from 'react';
+import { Component } from 'react';
 import './App.scss';
+import { Clock } from './components/Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type AppState = {
+  hasClock: boolean,
+  clockName: string,
+};
+
+export class App extends Component<{}, AppState> {
+  state = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
+
+  getClockNameId = 0;
```

Comment:
```suggestion
  clockNameIntervalId = 0;
```
use verbs (like `get`) in variable names for functions or methods